contrib/gorilla/mux/mux.go

```diff
@@ -96,18 +96,17 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	var (
-		match    mux.RouteMatch
-		spanopts []ddtrace.StartSpanOption
-		route    string
+		match mux.RouteMatch
+		route string
 	)
+	spanopts := options.Copy(r.config.spanOpts...)
 	// get the resource associated to this request
 	if r.Match(req, &match) && match.Route != nil {
 		if h, err := match.Route.GetHostTemplate(); err == nil {
 			spanopts = append(spanopts, tracer.Tag("mux.host", h))
 		}
 		route, _ = match.Route.GetPathTemplate()
 	}
-	spanopts = append(spanopts, r.config.spanOpts...)
```

Comment:
This one was ok already. Doesn't modify `spanOpts`.

Reply:
But it creates a handler that relies on `spanOpts`, that could be changed outside this function, right? (like https://github.com/DataDog/dd-trace-go/pull/2418#discussion_r1421043051)

Reply:
We just can't modify `r.config.spanOpts`. This copies `spanOpts` into a new array (old line 100).

Since we're also not creating a state closure, we only hold onto this as long as `ServeHTTP` runs, so there's no point making a copy here to protect ourselves.

Reply:
Talked offline. It's fine as it is now.